### PR TITLE
Add new-window API to expose WebDriver's New Window endpoint

### LINKED ISSUE #679: Implement WebDriver New Window function
**Problem/Opportunity**
Today, Etaoin has functions to switch among browser windows and close them, but it does not implement the WebDriver New Window function to create them.

**Proposed Solution**
Implement a `new-window` function that is based on the WebDriver New Window endpoint.

**Alternative Solutions**
None. There is no way to create new windows other than the load specific HTML into the browser that opens a link in a new window/frame and to click on that link. In fact, that's what the current Etaoin test suite does to create a new window so that it can validate that it can switch between windows correctly.

**Additional context**
None.

**Action**
I can submit a PR.

### PR DESCRIPTION
**Pull request by dgr**
Closes #679 

- [x] I have read [CONTRIBUTING](https://github.com/clj-commons/etaoin/blob/master/CONTRIBUTING.md) and the [Etaoin Developer Guide](https://github.com/clj-commons/etaoin/blob/master/doc/02-developer-guide.adoc).

- [x] This PR corresponds to an issue that the Etaoin maintainers have agreed to address. 

- [x] This PR contains test(s) to protect against future regressions

- [x] I have updated [CHANGELOG.adoc](https://github.com/clj-commons/etaoin/blob/master/CHANGELOG.adoc) with a description of the addressed issue.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ on:
   pull_request:
     branches:
     - master
+  workflow_dispatch: # for manual triggering
```
Comment (lread) [inline .github/workflows/test.yml:11]: Dave, this is unrelated to your change; did you include this on purpose?

File: CHANGELOG.adoc [hunk 2]
```diff
@@ -19,6 +19,9 @@ A release with an intentional breaking changes is marked with:
 //   (adjust these in publish.clj as you see fit)
 == Unreleased
 
+* Changes
+** {issue}679[#679]: Add `new-window` function that exposes WebDriver's New Window endpoint.
+
```
Comment (lread) [inline CHANGELOG.adoc:24]: Don't forget to credit yourself!

File: test/etaoin/api_test.clj [hunk 3]
```diff
@@ -637,6 +637,16 @@
         (is (not= width width'))
         (is (not= height height'))))))
 
+(deftest test-new-window
+  (is (= 1 (count (e/get-window-handles *driver*))))
+  (let [initial-window (e/get-window-handle *driver*)
+        new-windows (for [_ (range 3)]
+                      (-> (e/new-window *driver* :tab)
```
Comment (lread) [inline test/etaoin/api_test.clj:644]: I'm guessing you aren't also checking `:window` (and return `:type`) because browser support varies?

**Review by lread — CHANGES_REQUESTED** [review]
Hey Dave, looking good, I have a few comments.

File: .github/workflows/test.yml (hunk 1 above)
Comment (dgr) [reply .github/workflows/test.yml:11]: Shoot, that was a leftover from trying to debug the crashes. Let me remove that.

File: CHANGELOG.adoc (hunk 2 above)
Comment (dgr) [reply CHANGELOG.adoc:24]: Will do.

File: test/etaoin/api_test.clj (hunk 3 above)
Comment (lread) [reply test/etaoin/api_test.clj:644]: Coolio, tx for explaining!
